app/controllers/course/courses_controller.rb

```diff
@@ -6,9 +6,10 @@ def new #:nodoc:
   end
 
   def create #:nodoc:
-    @course.course_users.build(user: current_user, name: 'name', role: :owner)
+    creator = @course.course_users.build(user: current_user, name: 'name', role: :owner)
 
     if @course.save
+      creator.approve!
```

Comment:
I didn't see their official API about this, but we can directly do <code>build(workflow_state: 'approved')</code>